sklearn/neighbors/classification.py

```diff
@@ -383,10 +408,111 @@ def predict(self, X):
 
             y_pred[inliers, k] = classes_k.take(mode)
 
-        if outliers:
-            y_pred[outliers, :] = self.outlier_label
+            if outliers:
+                if self.outlier_label == 'most_frequent':
+                    prior = np.bincount(_y[:, k])
+                    frequent_label_index = prior.argmax()
+                    y_pred[outliers, k] = classes_k[frequent_label_index]
 
+        if outliers and isinstance(self.outlier_label, (numbers.Integral,
+                                                        np.integer)):
+            y_pred[outliers, :] = self.outlier_label
+            warns.warn('No neighbors found for test samples %r, '
+                       'their labels will be assgined with %d. '
+                       % (outliers, self.outlier_label))
         if not self.outputs_2d_:
             y_pred = y_pred.ravel()
 
         return y_pred
+
+    def predict_proba(self, X):
+        """Return probability estimates for the test data X.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_query, n_features), \
+                or (n_query, n_indexed) if metric == 'precomputed'
+            Test samples.
+
+        Returns
+        -------
+        p : array of shape = [n_samples, n_classes], or a list of n_outputs
+            of such arrays if n_outputs > 1.
+            The class probabilities of the input samples. Classes are ordered
```

Comment:
Classes must be ordered according to classes_

Reply:
This description is copied from the predict_proba() of KNeighborsClassifier. I think it's OK to use here since they have the same logic.